opensearch/src/main/java/org/opensearch/sql/opensearch/request/InitialPageRequestBuilder.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.opensearch.request;
+
+import static org.opensearch.sql.opensearch.request.OpenSearchRequestBuilder.DEFAULT_QUERY_TIMEOUT;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import lombok.Getter;
+import org.apache.commons.lang3.tuple.Pair;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.search.aggregations.AggregationBuilder;
+import org.opensearch.search.builder.SearchSourceBuilder;
+import org.opensearch.search.sort.SortBuilder;
+import org.opensearch.sql.ast.expression.Literal;
+import org.opensearch.sql.common.setting.Settings;
+import org.opensearch.sql.data.type.ExprType;
+import org.opensearch.sql.expression.ReferenceExpression;
+import org.opensearch.sql.opensearch.data.value.OpenSearchExprValueFactory;
+import org.opensearch.sql.opensearch.response.agg.OpenSearchAggregationResponseParser;
+
+public class InitialPageRequestBuilder implements PagedRequestBuilder {
+
+  @Getter
+  private final OpenSearchRequest.IndexName indexName;
+  private final SearchSourceBuilder sourceBuilder;
+  private final OpenSearchExprValueFactory exprValueFactory;
+  private final int querySize;
+
+  /**
+   * Constructor.
+   * @param indexName index being scanned
+   * @param settings other settings
+   * @param exprValueFactory value factory
+   */
+  // TODO accept indexName as string (same way as `OpenSearchRequestBuilder` does)?
```

Comment:
TODO
is that beneficial?

Reply:
Different builders accept different shapes of index name. Probably we need to make a standard for that.